test/test.py

```diff
@@ -4,32 +4,62 @@
 import subprocess
 import sys
 
+if "CRUNCH_BUILD_DIR" in os.environ.keys():
+    build_dir = os.environ["CRUNCH_BUILD_DIR"]
+else:
+    build_dir = "build"
+
+if "CRUNCH_RUNNER" in os.environ.keys():
+    crunch_runner = os.environ["CRUNCH_RUNNER"]
+else:
+    crunch_runner = None
+
+if "CRUNCH_EXTENSION" in os.environ.keys():
+    crunch_extension = os.environ["CRUNCH_EXTENSION"]
+else:
+    crunch_extension = None
+
 def print_command(command_list):
     print("running: " + " ".join(command_list), file=sys.stderr)
 
 def convert_path(path):
+    if path.startswith("build/"):
+        path = build_dir + path[len("build"):]
     return path.replace("/", os.path.sep)
 
 def run(command_list):
+    if crunch_runner:
+        command_list = crunch_runner.split(" ") + command_list
     print_command(command_list)
     returncode = subprocess.run(command_list).returncode
     if returncode:
         exit(returncode)
 
 def mkdir(path):
+    path = convert_path(path)
     print_command(["mkdir", path])
     os.makedirs(path, exist_ok=True)
 
-def crunch(input_path, output_path, options=[]):
-    executable_extension = ["", ".exe"][sys.platform == 'win32']
-    executable_name = "crunch" + executable_extension
-
-    build_dir = "build"
+def get_build_dir():
     windows_build_dir = os.path.join(build_dir, "Release")
     if os.path.exists(windows_build_dir):
-        build_dir = windows_build_dir
+        return windows_build_dir
+    return build_dir
 
-    executable_path = os.path.join(build_dir, executable_name)
+def get_executable_path(executable_name):
+    if crunch_extension:
+        executable_extension = crunch_extension
+    elif sys.platform == "win32":
```

Comment:
Not necessary in the native Windows case since `.exe` can be omitted from a command.

Reply:
If I'm right the Windows CI failed without, the error was that the file was not found, Python is likely executing the binary directly without passing the command to a shell.

I also need the explicit extension to test that code on Linux with Python doing `wine crunch.exe …`.

Reply:
> I also need the explicit extension […]

Well, this use case use the environment variable…

I don't think it's bad to be explicit anyway.

Reply:
So, it works without extension, and removing it also allows to delete multiple lines and a variable, so I remove that.